tests/integration-tests/conftest.py

```diff
@@ -1223,6 +1223,24 @@ def pytest_runtest_makereport(item, call):
             logging.error("Failed when generating config for failed tests: %s", e, exc_info=True)
 
 
+@pytest.fixture(scope="class")
+def serial_execution_by_instance(request, instance):
+    """Enforce serial execution of tests, according to the adopted instance."""
+    if instance in ["c5n.18xlarge", "p4d.24xlarge"]:
+        logging.info("Enforcing serial execution for instance %s", instance)
+        outdir = request.config.getoption("output_dir")
+        lock_file = f"{outdir}/{instance}.lock"
+        lock = FileLock(lock_file=lock_file)
+        logging.info("Acquiring lock file %s", lock.lock_file)
+        with lock.acquire(poll_interval=0.1):
```

Comment:
I think this poll interval is too strict, we're talking about a test execution so maybe we can check every 30seconds?

Then what about adding a timeout (2h) to the acquire method?

Reply:
Very good catch, thank you!